momepy/functional/_diversity.py

```diff
@@ -1,11 +1,101 @@
 import warnings
 
 import numpy as np
+import pandas as pd
 from libpysal.graph import Graph
 from numpy.typing import NDArray
 from pandas import DataFrame, Series
 
-__all__ = ["describe"]
+try:
+    from numba import njit
+except (ModuleNotFoundError, ImportError):
+    warnings.warn(
+        "The numba package is used extensively in this function to accelerate the"
+        " computation of statistics but it is not installed or  cannot be imported."
+        " Without numba, these computations may become slow on large data.",
+        UserWarning,
+        stacklevel=2,
+    )
+    from libpysal.common import jit as njit
```

Comment:
The warning should be printed only if a user calls a function that actually uses numba. That is why it was originally inside `describe`.  I would still keep the warning inside if possible.